src/js/ripple/log.web.js

```diff
@@ -8,21 +8,39 @@ var exports = module.exports = require('./log.js');
  * function without any stringification.
  */
 var InteractiveLogEngine = {
-  logObject: function (msg, obj) {
-    var args = Array.prototype.slice.call(arguments, 1);
+  logObject: function(level, message) {
+    var timestamp = '[' + new Date().toISOString() + ']';
+    var location = (new Error()).stack.split('\n')[4].replace(/^\s+/, '');
 
-    args = args.map(function(arg) {
-      if (/MSIE/.test(navigator.userAgent)) {
-        return JSON.stringify(arg, null, 2);
-      } else {
-        return arg;
-      }
+    var args = [
+      timestamp,
+      message,
+      '--',
+      location,
+      '\n'
+    ];
+
+    Array.prototype.slice.call(arguments, 2).forEach(function(arg) {
+      args.push(/MSIE/.test(navigator.userAgent)
```

Comment:
This is a lot of duplication for this one check. Would it be better to just always do this check so that we can use a single function?